0001-metric-pre-defined-labels.md

```diff
@@ -0,0 +1,37 @@
+# Pre-defined label support for all metric operations
+
+Let all Metric objects (Cumulative, Gauge, ...) and raw statistics support pre-defined label values.
+
+## Motivation
+
+In the current `Metric.GetOrCreateTimeSeries` API for Gauges and Cumulatives, the caller obtains a `TimeSeries` handle for repeatedly recording metrics with certain pre-defined label values set.  This is an important optimization, especially for exporting aggregated metrics.
+
+The use of pre-defined labels improves usability too, for working with metrics in code. Application programs with long-lived objects and associated Metrics can compute predefined label values once (e.g., in a constructor), rather than once per call site.
+
+The current API for recording raw statistics does not support the same optimization or usability advantage.  This RFC proposes to add support for pre-defined labels on all metrics.
+
+## Explanation
+
+In the current proposal, Metrics are used for _common_ pre-aggregated metric types, whereas Raw statistics are used for uncommon and vendor-specific aggregations.  The optimization and the usability advantages gained with pre-defined labels should be extended to Raw statistics because they equally important and equally applicable.
```

Comment:
```suggestion
In the current proposal, Metrics are used for _common_ pre-aggregated metric types, whereas Raw statistics are used for uncommon and vendor-specific aggregations.  The optimization and the usability advantages gained with pre-defined labels should be extended to Raw statistics because they are equally important and equally applicable.
```